src/test/java/org/terasology/math/geom/BaseVector2dTest.java

```diff
@@ -141,6 +141,40 @@ public void distanceSquared() {
         assertEquals(26, b.distanceSquared(a), EPSILON);
     }
 
+    @Test
+    public void project() {
+        BaseVector2d a = createBaseVector2d(-4, 3);
+        BaseVector2d b = createBaseVector2d(1, 2);
+
+        // a.lengthSquared() == 25.0
+        // a.dot(b) == 2
+        BaseVector2d projected = createBaseVector2d(-4 * 2 / 25.0, 3 * 2 / 25.0);
+
+        assertBaseVector2dEquals(projected, b.project(a), EPSILON);
+    }
+
+    @Test
+    public void projectAlongAxes() {
+        BaseVector2d a = createBaseVector2d(-4, 3);
```

Comment:
any reason for this intermediary vector? you are just using one of the components anyways. you might as well just plug in the value itself rather than create a vector.